core/src/databases/table.rs

```diff
@@ -228,41 +232,63 @@ impl LocalTable {
         &self,
         store: Box<dyn Store + Sync + Send>,
         databases_store: Box<dyn DatabasesStore + Sync + Send>,
-        rows: &Vec<Row>,
+        rows: Vec<Row>,
         truncate: bool,
     ) -> Result<()> {
-        // Validate that all rows keys are lowercase.
-        for (row_index, row) in rows.iter().enumerate() {
-            let object = match row.value().as_object() {
-                Some(object) => object,
-                None => Err(anyhow!("Row {} is not an object", row_index,))?,
-            };
-            match object.keys().find(|key| match key.chars().next() {
-                Some(c) => c.is_ascii_uppercase(),
-                None => false,
-            }) {
-                Some(key) => Err(anyhow!(
-                    "Row {} has a key '{}' that contains uppercase characters",
-                    row_index,
-                    key
-                ))?,
-                None => (),
-            }
+        let rows = Arc::new(rows);
+
+        let mut now = utils::now();
+        // Validate that all rows keys are lowercase. We run it in a spawn_blocking since it is CPU
```

Comment:
Why not rayon for this one ?

Reply:
This one is very fast already with wide tables

Reply:
or even deep tables... it's fast so no need for rayon but good hygiene to not block the pool